Initial scaffold for VA Form 21P-0537 (Marital Status Questionnaire for DIC Recipients)

amponce opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders


 ## Summary
  - Initial frontend scaffold for VA Form 21P-0537 (Marital Status
  Questionnaire for DIC Recipients)
  - Non-production implementation for DIC benefit recipients to report marital
  status changes
  - Form configuration with conditional logic for remarriage scenarios

  ## Changes
  - Add form configuration with chapters and pages structure
  - Implement prefill and submit transformers
  - Create React containers (App, Introduction, Confirmation pages)
  - Add form pages for recipient info, remarriage questions, marriage details,
  contact info, and certification
  - Set up routing and Redux reducers
  - Add comprehensive unit tests for submit transformer
  - Include test fixtures for various scenarios (remarried, not remarried, over
   57, terminated marriage)
  - Configure manifest and SASS styling

  ## Content Build Registry
  Content build registry PR:
  https://github.com/department-of-veterans-affairs/content-build/pull/2655
  
  To test locally:
  
  ```
  {
    "appName": "Marital Status Questionnaire for DIC Recipients",
    "entryName": "21p-0537",
    "rootUrl": "/supporting-forms-for-claims/marital-status-questionnaire-21p-0537",
    "template": {
      "title": "Marital Status Questionnaire for DIC Recipients",
      "layout": "page-react.html",
      "vagovprod": false,
      "includeBreadcrumbs": true,
      "breadcrumbs_override": [
        {
          "path": "supporting-forms-for-claims",
          "name": "Supporting forms for VA claims"
        },
        {
          "path": "supporting-forms-for-claims/marital-status-questionnaire-21p-0537",
          "name": "Marital Status Questionnaire for DIC Recipients"
        }
      ]
    }
  }
```

  ## Test Plan
  - [x] Unit tests for submit transformer with comprehensive coverage
  - [x] Test fixtures for all form scenarios
  - [x] Form configuration validates correctly
  - [ ] E2E testing (will be added in future PRs)
  - [ ] Accessibility testing (will be added in future PRs)

  ## Notes
  - This is initial scaffold work - not production ready
  - Conditional logic implemented for skipping sections based on remarriage
  status
  - Form supports both SSN and VA file number identification




## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#120257


## Testing done

  - [x] Unit tests for submit transformer with comprehensive coverage
  - [x] Test fixtures for all form scenarios
  - [x] Form configuration validates correctly
  - [ ] E2E testing (will be added in future PRs)
  - [ ] Accessibility testing (will be added in future PRs)
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         |  _ |  _ |
| ------- | ------ | ----- |
| Desktop |  <img width="1118" height="3224" alt="localhost_3001_supporting-forms-for-claims_marital-status-questionnaire-21p-0537_introduction" src="https://github.com/user-attachments/assets/21bc9007-88ac-4169-971c-ee6131b92b00" />      |       <img width="1118" height="5212" alt="localhost_3001_supporting-forms-for-claims_marital-status-questionnaire-21p-0537_introduction (1)" src="https://github.com/user-attachments/assets/ca648d0c-664e-46cb-b671-cd1e74167fee" />|

## What areas of the site does it impact?

/supporting-forms-for-claims/marital-status-questionnaire-21p-0537/review-and-submit

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria
- Initial form is scaffolded and visible on staging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

michaelclement commented on src/applications/manifest-catalog.json:
```diff
@@ -141,12 +141,6 @@
       "entryName": "claims-status",
       "rootUrl": "/track-claims"
     },
-    {
-      "directoryPath": "src/applications/claims-status/node_modules/@stencil/core/screenshot/compare",
-      "appName": "N/A",
-      "entryName": "N/A",
-      "rootUrl": "N/A"
-    },
```
Should this change be included? Looks possibly like an unintentional removal

amponce replied:
ohh, nope that was an artifact from the scaffold tool, thanks for catching that

michaelclement reviewed (dismissed):
Pulled it down and walked through it locally, looks good and functioned as expected. LGTM 👍

amponce committed:
feat: Add form 21P-0537 Marital Status Questionnaire for DIC Recipients

- Initial scaffold for form 21P-0537
- Add form registration in platform constants
- Configure save-in-progress functionality
- Add form to manifest catalog
- Fix ConfirmationPage alert display
- Add validation exemption for missing vets-json-schema

amponce commented on src/applications/manifest-catalog.json:
```diff
@@ -385,7 +379,7 @@
       "appName": "File a CHAMPVA claim",
       "entryName": "10-7959a",
-      "rootUrl": "/family-and-caregiver-benefits/health-and-disability/file-champva-claim-10-7959a"
+      "rootUrl": "/family-and-caregiver-benefits/health-and-disability/champva/file-champva-claim-10-7959a"
```
It looks like this is getting auto generated

michaelclement replied:
I’m curious if it’s being generated from outdated local info? Definitely seems strange that another team’s info would be changed by our form addition

michaelclement reviewed (approved):
Ran locally, everything still looks good - thanks for updating the registration. This'll be great to have in staging. 🚢
